### PR TITLE
Forking uses passed model rather than stale context

### PR DESCRIPTION
**Pull request by Rheeseyb**
**Problem:**
The forking process in our persistence state machine uses the version of the project that is stored in the machine's context rather than the model that was passed in when asking the machine to save the model, meaning there is an edge case where auto-forking an old project could fail if migration steps were required.

**Fix:**
Use the model that was passed in as part of the save event when performing the fork. I've also added a test to ensure that the fork uses the latest version of the project.

### CODE REVIEW

**Review by gbalint — APPROVED** [review]

File: editor/src/components/editor/persistence/generic/persistence-machine.ts [hunk 1]
```diff
@@ -743,6 +748,7 @@ export function createPersistenceMachine<ModelType, FileType>(
                         },
                       }),
                     },
+                    // FIXME Error handling
```
Comment (seanparsons) [inline editor/src/components/editor/persistence/generic/persistence-machine.ts:751]: Is this what I saw you (elsewhere) comment on that some further fix was needed?

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/editor/persistence/generic/persistence-machine.ts (hunk 1 above)
Comment (Rheeseyb) [reply editor/src/components/editor/persistence/generic/persistence-machine.ts:751]: Yeah, I have a follow up PR I'll be opening in a sec that deals with this
